File: Searching/Majority Element.py
def find_majority(arr):
    res, count = 0, 1
    n = len(arr)
    for i in range(1, n):
        if arr[res] == arr[i]:
            count += 1
        else:
            count -= 1
        if count == 0:
            res = i
            count = 1
    count = 0
    for i in range(n):
        if arr[i] == arr[res]:
            count += 1
    if count <= n//2:
        return -1
    return res

arr = list(map(int, input('Enter array: ').split()))
print(f'Majority element in array is {find_majority(arr)}')

File: Array/MajorityElement.java
package Array;

public class MajorityElement {

  public static int findMajority(int[] arr) {
    int resIndex = 0;
    int count = 1;
    for (int i = 1; i < arr.length; i++) {
      if (arr[resIndex] == arr[i])
        count++;
      else
        count--;
      if (count == 0) {
        resIndex = i;
        count = 1;
      }
    }
    count = 0;
    for (int i = 0; i < arr.length; i++) {
      if (arr[resIndex] == arr[i])
        count++;
    }
    if (count <= arr.length/2)
      resIndex = -1;
    return resIndex;
  }

  public static void main(String[] args) {
    int[] arr = Basic.inputArray();
    System.out.println("Majority Element is at index " + findMajority(arr));
  }
  
}
